app/services/memberships/revoke_service.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Memberships
+  class RevokeService < BaseService
+    def revoke(id)
+      membership = Membership.find_by(id: id)
+      return result.fail!(code: 'not_found') unless membership
+
+      if result.user.id == membership.user.id
+        return result.fail!(
+          code: 'unprocessable_entity',
+          message: 'revoke_own_membership',
```

Comment:
```suggestion
          message: 'trying to revoke my own membership',
```

I think message is more a sentence than a code in the codebase

Reply:
Hum indeed, I added a message first but I also wanted to send a detail with the code `revoke_own_membership`

```
return result.fail!(
          code: 'unprocessable_entity',
          message: 'Cannot revoke own membership',
          detail: 'revoke_own_membership',
)
```

The issue I had is that the `detail` was never received by the client, like if some kind of serializer didn't allowed such attribute. So I changed the `message` value to have the value of `detail` 😅

Reply:
Updated